addons/mail/static/src/new/thread/thread.js

```diff
@@ -16,6 +16,20 @@ export class Thread extends Component {
                     !this.props.messageHighlight.highlightedMessageId
             );
         }
+        this.loadMoreState = useVisible("load-more");
+        useEffect(
+            (isVisible) => {
+                if (isVisible) {
+                    this.messaging.fetchThreadMessagesMore(this.props.id);
+                }
+            },
+            () => [this.loadMoreState.isVisible]
+        );
+        onPatched(() => {
+            if (this.loadMoreState.isVisible) {
+                this.messaging.fetchThreadMessagesMore(this.props.id);
+            }
+        });
```

Comment:
Why both `useEffect` and `onPatched`?

Reply:
And how comes we base the load more based on lifecycle?
If the action of "viewing" the load more button does not trigger a render (and why should it?) then it looks like these methods won't be called.

Reply:
Removed `useEffect` and added `onMounted`.
`useVisible` returns a reactive whose value depends on visibility of the load more button.
So it triggers a re-render whenever the value changes.
The auto-scroll on load is necessarily linked to visibility of load more button.